app/src/main/java/com/example/concordia_campus_guide/LocationFragment/LocationFragmentViewModel.java

```diff
@@ -91,49 +94,69 @@ private void setPolygonStyle(GeoJsonLayer layer, GoogleMap map){
 
             String[] coordinates = feature.getProperty("center").split(", ");
             LatLng centerPos = new LatLng(parseDouble(coordinates[1]), parseDouble(coordinates[0]));
-            addBuildingMarker(map, centerPos, Enum.valueOf(BuildingCode.class, feature.getProperty("code")));
+            addBuildingMarker(map, centerPos, feature.getProperty("code"), context);
         }
```

Comment:
this is not specifically for you or your code but I noticed we used these methods a lot to get code properties such as code height, center, etc. maybe we should extract this in a method where we call it and get it directly, seems like duplicate codes. I thought we might want to have a building object created with all those info and then we just call the getters